server/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -500,6 +506,58 @@ private void startLookups(List<LookupBean> lookupBeanList)
     }
   }
 
+  /**
+   * @return a pair of the map with successful lookups and the list of failed lookups
+   */
+  private Pair<Map<LookupBean, LookupExtractorFactoryContainer>, List<LookupBean>> startLookups(
+      List<LookupBean> lookupBeans,
+      CompletionService<Map.Entry<LookupBean, LookupExtractorFactoryContainer>> completionService,
+      List<Future<?>> allFutures
+  ) throws InterruptedException
+  {
+    for (LookupBean lookupBean : lookupBeans) {
+      allFutures.add(completionService.submit(() -> startLookup(lookupBean)));
+    }
+    Map<LookupBean, LookupExtractorFactoryContainer> successfulLookups = new HashMap<>();
+    for (int i = 0; i < lookupBeans.size(); i++) {
+      Future<Map.Entry<LookupBean, LookupExtractorFactoryContainer>> completedFuture = completionService.take();
+      try {
+        Map.Entry<LookupBean, LookupExtractorFactoryContainer> lookupResult = completedFuture.get();
+        if (lookupResult != null) {
+          successfulLookups.put(lookupResult.getKey(), lookupResult.getValue());
+        }
+      }
+      catch (ExecutionException e) {
+        // not adding to successfulLookups
```

Comment:
This seems like the appropriate place to log an error since it is where the error is ignored.

Reply:
added